source/SupportLib/PlayGamesPluginSupport/src/main/java/com/google/games/bridge/ShowInboxUiRequest.java

```diff
@@ -0,0 +1,91 @@
+package com.google.games.bridge;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.util.Log;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.common.api.CommonStatusCodes;
+import com.google.android.gms.games.Games;
+import com.google.android.gms.games.GamesActivityResultCodes;
+import com.google.android.gms.games.multiplayer.Multiplayer;
+import com.google.android.gms.games.multiplayer.turnbased.TurnBasedMatch;
+import com.google.android.gms.games.TurnBasedMultiplayerClient;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.TaskCompletionSource;
+
+
+class ShowInboxUiRequest implements HelperFragment.Request {
+    private static final String TAG = "SimpleUiRequest";
+
+    private final TaskCompletionSource<Result> resultTaskSource = new TaskCompletionSource<>();
+
+    public class Result {
+        public int status;
+        public TurnBasedMatch turnBasedMatch;
+
+        Result(int status, TurnBasedMatch turnBasedMatch) {
+            this.status = status;
+            this.turnBasedMatch = turnBasedMatch;
+        }
+    }
+
+    public Task<Result> getTask() {
+        return resultTaskSource.getTask();
+    }
+
+    public void process(final HelperFragment helperFragment) {
+        final Activity activity = helperFragment.getActivity();
+        GoogleSignInAccount account = HelperFragment.getAccount(activity);
+        TurnBasedMultiplayerClient client = Games.getTurnBasedMultiplayerClient(activity, account);
+        client.getInboxIntent()
+            .addOnSuccessListener(
+                activity,
+                    new OnSuccessListener<Intent>() {
+                        @Override
+                        public void onSuccess(Intent intent) {
+                            helperFragment.startActivityForResult(intent, HelperFragment.RC_SHOW_INBOX_UI);
+                        }
+                    })
+            .addOnFailureListener(
+                activity,
+                    new OnFailureListener() {
```

Comment:
> new OnFailureListener() {

Should be on the same level as activity param

Reply:
Fixed this one and others in the same directory.